File: example/java/jimmer-core/src/main/java/org/babyfish/jimmer/example/core/model/Book.java
package org.babyfish.jimmer.example.core.model;

import org.babyfish.jimmer.Immutable;
import org.jetbrains.annotations.Nullable;

import javax.validation.constraints.Null;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;
import java.util.List;

@Immutable
public interface Book {

    @Size(max = 50)
    String name();

    @Null // Nullable property, Java-API needs it, but kotlin-API does not.
    BookStore store();

    int price();

    LocalDateTime lastModifiedTime();

    List<Author> authors();
}

File: example/kotlin/jimmer-core-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/core/model/Book.kt
package org.babyfish.jimmer.example.kt.core.model

import org.babyfish.jimmer.Immutable
import java.time.LocalDateTime
import javax.validation.constraints.Size

@Immutable
interface Book {

    @get:Size(max = 50)
    val name: String

    val store: BookStore?

    val price: Int

    val lastModifiedTime: LocalDateTime

    val authors: List<Author>
}